### PR TITLE
Refactor: use version instead of major version in windows build

### LINKED ISSUE #49219: Refactor: Don't specify CUDA complete version manually in Windows build scripts.
## 🚀 Feature
Don't specify CUDA complete version manually in Windows build script.

## Motivation

So far, we have to update the CUDA complete version manually in several scripts. It increases the possibility of making mistakes.
For example, we take a lot time to debug https://github.com/pytorch/pytorch/pull/47794.
And yesterday, I find we're  still using magma 110 to build CUDA 11.1.
https://github.com/pytorch/pytorch/blob/59e822026c4ddbbdc26266f0fccdd14393098c42/.jenkins/pytorch/win-test-helpers/installation-helpers/install_magma.bat#L3-L12

## Pitch

 It'd better to pass the right cuda_compelete_version to build scripts.

cc @malfet @seemethere @walterddr @peterjc123 @maxluk @nbcsm @guyang3532 @gunandrose4u @mszhanyi @skyline75489

### PR DESCRIPTION
**Pull request by mszhanyi**
Fixes #49219

1. update version instead of major version for env var of CUDA_VERSION
2. update related scripts

### CODE REVIEW

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat [hunk 1]
```diff
@@ -37,33 +37,19 @@ if "%VC_VERSION%" == "" (
...
+if not "%USE_CUDA%"=="1" goto cuda_build_end
 
-:cuda_build_9
+set CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v%CUDA_COMPLETE_VERSION%
 
-set CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.2
-set CUDA_PATH_V9_2=%CUDA_PATH%
+rem version transformer, for example 10.1 to 10_1. But I don't find where the CUDA_PATH_VXX_X is used 
+set VERSION_SUFFIX=%CUDA_COMPLETE_VERSION:.=_%
+set CUDA_PATH_V%VERSION_SUFFIX%=CUDA_PATH
```
Comment (peterjc123) [inline .jenkins/pytorch/win-test-helpers/build_pytorch.bat:46]: Well, are you sure this line is correct?

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat [hunk 2]
```diff
@@ -37,33 +37,19 @@ if "%VC_VERSION%" == "" (
...
-set CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.1
-set CUDA_PATH_V11_1=%CUDA_PATH%
-
-goto cuda_build_common
-
-:cuda_build_common
+set CUDNN_LIB_DIR=%CUDA_PATH%\lib\x64
+set CUDA_TOOLKIT_ROOT_DIR=%CUDA_PATH%
+set CUDNN_ROOT_DIR=%CUDA_PATH%
+set NVTOOLSEXT_PATH=C:\Program Files\NVIDIA Corporation\NvToolsExt
```
Comment (peterjc123) [inline .jenkins/pytorch/win-test-helpers/build_pytorch.bat:51]: Some of the lines have repeated more than once. Could you please do a cleanup?

**Review by peterjc123 — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -326,6 +326,9 @@ pytorch_windows_params: &pytorch_windows_params
     cuda_version:
       type: string
       default: "10"
+    cuda_complete_version:
```
Comment (peterjc123) [inline .circleci/config.yml:329]: BTW, I don't think CUDA COMPLETE VERSION is good name here. The correct names and the corresponding values:
cuda version: 10.1
cuda major version: 10
cuda complete version 10.1.249

**Comment by peterjc123** [comment]
> 3\. Could we remove the env like CUDA_PATH_V10_1?

No, they are set when you install CUDA manually. As you can see those variables related to `CUDA_PATH` in my PC.
```cmd
C:\Users\peter>set | findstr "CUDA_PATH"
CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1
CUDA_PATH_V10_1=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1
CUDA_PATH_V10_2=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.2
CUDA_PATH_V11_0=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.0
```
BTW, they are also used when trying to import `torch`. I don't get why you would like to do that.

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat (hunk 1 above)
Comment (mszhanyi) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat:46]: Good Catch. it should _set CUDA_PATH_V%VERSION_SUFFIX%=%CUDA_PATH%_

File: .circleci/config.yml (hunk 3 above)
Comment (mszhanyi) [reply .circleci/config.yml:329]: Your comments is valuable. In some scripts,  the name of cuda_version is used as cuda major version. Let me think if I should update them all in this PR.

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat (hunk 2 above)
Comment (mszhanyi) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat:51]: do you mean it's duplicated of https://github.com/pytorch/pytorch/blob/5b6b1495b9bc27ddb2cf10df6a387090716eb16a/.jenkins/pytorch/win-test-helpers/setup_pytorch_env.bat#L77-L84

Comment (peterjc123) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat:51]: Yeah.

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat (hunk 1 above)
Comment (mszhanyi) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat]: https://circleci.com/api/v1.1/project/github/pytorch/pytorch/9567605/output/104/0?file=true&allocation-id=5fd480fbf1f454400cd69497-0-build%2F12B49615

```
(base) circleci@PACKER-5FD0AAF5 C:\Users\circleci\project>rem version transformer, for example 10.1 to 10_1. 
(base) circleci@PACKER-5FD0AAF5 C:\Users\circleci\project>set VERSION_SUFFIX=10_1 
(base) circleci@PACKER-5FD0AAF5 C:\Users\circleci\project>set CUDA_PATH_V10_1=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1 
(base) circleci@PACKER-5FD0AAF5 C:\Users\circleci\project>set CUDNN_LIB_DIR=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\lib\x64
```

File: .jenkins/pytorch/win-test-helpers/build_pytorch.bat (hunk 2 above)
Comment (peterjc123) [reply .jenkins/pytorch/win-test-helpers/build_pytorch.bat:51]: Repeated variable definitions:
NVTOOLSEXT_PATH: Line 51, 57
CUDA_TOOLKIT_ROOT_DIR: Line 49, 55
CUDNN_ROOT_DIR: Line 50, 56
CUDNN_LIB_DIR: Line 48, 54

File: .jenkins/pytorch/win-test-helpers/installation-helpers/install_magma.bat [hunk 4]
```diff
@@ -1,9 +1,9 @@
-if "%CUDA_VERSION%" == "9" set CUDA_SUFFIX=cuda92
-if "%CUDA_VERSION%" == "10" set CUDA_SUFFIX=cuda101
-if "%CUDA_VERSION%" == "11" set CUDA_SUFFIX=cuda110
+rem remove dot in cuda_version, fox example 11.1 to 111
+set VERSION_SUFFIX=%CUDA_VERSION:.=%
+set CUDA_SUFFIX=cuda%VERSION_SUFFIX%
 
 if "%CUDA_SUFFIX%" == "" (
-  echo unknown CUDA version, please set `CUDA_VERSION` to 9, 10 or 11.
+  echo unknown CUDA version, please set `CUDA_COMPLETE_VERSION` higher than 9.2
```
Comment (peterjc123) [inline .jenkins/pytorch/win-test-helpers/installation-helpers/install_magma.bat:6]: Should be `CUDA_VERSION`, right?

Comment (mszhanyi) [reply .jenkins/pytorch/win-test-helpers/installation-helpers/install_magma.bat:6]: Thanks，updated

File: .jenkins/pytorch/win-test-helpers/setup_pytorch_env.bat [hunk 5]
```diff
@@ -55,11 +55,6 @@ rem version transformer, for example 10.1 to 10_1.
 set VERSION_SUFFIX=%CUDA_VERSION:.=_%
 set CUDA_PATH_V%VERSION_SUFFIX%=%CUDA_PATH%
 
-set CUDNN_LIB_DIR=%CUDA_PATH%\lib\x64
-set CUDA_TOOLKIT_ROOT_DIR=%CUDA_PATH%
-set CUDNN_ROOT_DIR=%CUDA_PATH%
-set NVTOOLSEXT_PATH=C:\Program Files\NVIDIA Corporation\NvToolsExt
-set PATH=%CUDA_PATH%\bin;%CUDA_PATH%\libnvvp;%PATH%
```
Comment (peterjc123) [inline .jenkins/pytorch/win-test-helpers/setup_pytorch_env.bat:81]: Please revert the change on this line.

Comment (mszhanyi) [reply .jenkins/pytorch/win-test-helpers/setup_pytorch_env.bat]: Done

**Review by peterjc123 — APPROVED** [review]

File: .circleci/scripts/windows_cuda_install.sh [hunk 6]
```diff
@@ -16,7 +14,7 @@ else
     exit 1
 fi
 
-if [[ "${CUDA_VERSION}" != "10" && "${JOB_EXECUTOR}" == "windows-with-nvidia-gpu" ]]; then
+if [[ "$CUDA_VERSION" =~ ^10.* && "${JOB_EXECUTOR}" == "windows-with-nvidia-gpu" ]]; then
```
Comment (malfet) [inline .circleci/scripts/windows_cuda_install.sh:17]: This introduced runtime failures in CI, because it should be `"$CUDA_VERSION" =~ ^11.*`

Comment (mszhanyi) [reply .circleci/scripts/windows_cuda_install.sh:17]: Thank your help. So sorry for that.
